### ISSUE TITLE
Modifying an rvalue reference of a string literal causes segfault

### ISSUE THREAD

**Issue by marler8997** [ziglang/zig#3444]
Take the following example:
```zig
fn popFirstElement(stringRef: *[]const u8) void {
    stringRef.* = stringRef.*[1..];
}

test "bug" {
    popFirstElement(&"foo"[0..]);
}
```

The `&"foo"[0..]` expression is taking the address of an rvalue.  However, when it is modified in the `popFirstElement` function this segfault occurs:
```
1/1 test "bug"...Segmentation fault at address 0x201578
/home/marler8997/git/zog/bug.zig:2:30: 0x205723 in popFirstElement (test)
    stringRef.* = stringRef.*[1..];
                             ^
/home/marler8997/git/zog/bug.zig:6:20: 0x205592 in test "bug" (test)
    popFirstElement(&"foo"[0..]);
                   ^
/home/marler8997/bin/zig-linux-x86_64-0.5.0+8b459216/lib/zig/std/special/test_runner.zig:13:25: 0x2284e1 in std.special.main (test)
        if (test_fn.func()) |_| {
                        ^
/home/marler8997/bin/zig-linux-x86_64-0.5.0+8b459216/lib/zig/std/special/start.zig:204:37: 0x227355 in std.special.posixCallMainAndExit (test)
            const result = root.main() catch |err| {
                                    ^
/home/marler8997/bin/zig-linux-x86_64-0.5.0+8b459216/lib/zig/std/special/start.zig:102:5: 0x2271cf in std.special._start (test)
    @noInlineCall(posixCallMainAndExit);
    ^

Tests failed. Use the following command to reproduce the failure:
/home/marler8997/git/zog/zig-cache/o/lNWDEb7CbHCUpRDUVTsn8ZiwsvJwp9_hF8wwgR91Xl4TNulOrqSQPw1Tzn-dT9B_/test
```

I'm not sure how rvalue references work in Zig, but I'm guessing that they are supposed to be const. In this case this would mean that the expression `&"foo"[0..]` should become type `*const []const u8` rather than `*[]const u8`.  If this was the case, then this would have been a compile error because you wouldn't be able to pass a const reference to the `popFirstElement` function.

**Comment by marler8997** [comment]
Curiously, this seems to work with integer values.  See this example:
```zig
fn setZero(numberRef: *u8) void {
    numberRef.* = 0;
}

fn return42() u8 { return 42; }

test "bug" {
   setZero(&return42());
}
```

**Comment by marler8997** [comment]
Actually this also works if you don't use a string literal rvalue, but use a function return rvalue instead:
```zig
fn popFirstElement(stringRef: *[]const u8) void {
    stringRef.* = stringRef.*[1..];
}

fn returnString() []const u8 {
    return "foo";
}

test "bug" {
    popFirstElement(&returnString());
}
```

**Comment by Vexu** [comment]
Duplicate of #2130, fixed by #4971.
```
./build/a.zig:279:27: error: expected type '*[]const u8', found '*const *const [3:0]u8'
    popFirstElement(&"foo"[0..]);
                          ^
./build/a.zig:279:27: note: cast discards const qualifier
    popFirstElement(&"foo"[0..]);
                          ^
```
